Show 'What's New' popup on first start after upgrade to a new version (#895)
databricks/databricks-vscode: packages/databricks-vscode/resources/whats-new/1.2.md

> # <!-- DATABRICKS LOGO --> Databricks Extension v1.2

Reviewer: Missing patch number. Would this be v1.2.0?

Reviewer: Should probably be full name: "Databricks extension for Visual Studio Code"

Author: Patch number is intentionally missing, since this should show up when users cross the minor boundary, for any patch versions. How about v1.2.x?

## What's new?

> -   [📗 Advanced notebook support with Databricks Connect V2](#dbconnect-notebook)

Reviewer: We really should not use V2 in our docs. It was an internal thing so that we could distinguish it from V1 at first. But now that it has escaped into the wild, a few customers have been complaining about where to find "v2.0.0" of Databricks Connect, which doesn't actually exist. Instead, we've been using something like "Databricks Connect for Databricks Runtime 13.0 and above" on first mention, and then once that context is clear, shortening simply to "Databricks Connect" on subsequent mentions. You can also keep the short version of "Databricks Connect" in titles, with the first mention in the body being "Databricks Connect for Databricks Runtime 13.0 and above" for clarity.

> -   [📗 Advanced notebook support with Databricks Connect V2](#dbconnect-notebook)

Reviewer: ```suggestion
-   [📗 Advanced notebook support with Databricks Connect](#dbconnect-notebook)
```

> ## <a id="dbconnect-notebook"></a> Advanced notebook support with Databricks Connect V2

Reviewer: ```suggestion
## <a id="dbconnect-notebook"></a> Advanced notebook support with Databricks Connect
```

> You could already run notebooks locally using Databricks Connect V2. But we are now introducing advanced notebook support which enables you to run notebooks locally with the similar experience as running them on a Databricks cluster. The aim of this update is to bring most of the convenience of running notebooks on a Databricks cluster to your local environment!

Reviewer: ```suggestion
Before this update, you could already run notebooks locally with Databricks Connect. This update introduces advanced notebook support. This advanced support enables you to run notebooks locally with a similar experience as you have by running them on a Databricks cluster. The goal of this update is to bring much of the convenience of running notebooks on a Databricks cluster to your local development environment.
```

> This includes:
> 
> -   [Magic commands such as `%run` and `%sql` etc](#magic-commands)
> -   [Preconfigured globals such as spark and dbutils](#preconf-globals)
> -   [Interactive `dbutils.widgets`](#widgets)
> -   [Support for Databricks Notebooks](#dbnb)

Reviewer: ```suggestion
This update includes:

-   [Magic commands such as `%run` and `%sql`](#magic-commands)
-   [Preconfigured globals such as `spark` and `dbutils`](#preconf-globals)
-   [Interactive widgets through `dbutils.widgets`](#widgets)
-   [Support for Databricks notebooks](#dbnb)
```

Reviewer: `spark` and `dbutils` are programmatic constructs, hence the backticks. 

"notebooks" should not be capitalized.

### <a id="magic-command"></a>Magic commands

You can now use Databricks magic commands locally from your notebooks.

> See [here]() for a list of supported magic commands.

Reviewer: As a general rule, you should never use `[here]` as link text, here and throughout. This is problematic for accessibility, and it does not help the user confirm that they got to the right target. Instead, try to copy or approximate the title of the target page or article. 

Since I'm not sure what you're specifically trying to link to, I'm not sure how to propose appropriate link text edits here and throughout.

Reviewer: Also, you should lead with the "for" statement here and throughout, to help short-circuit readers about whether they need to bother clicking the link. For example: "For a list of supported magic commands, see `[link-text](url)`."

> See [here]() for a list of supported magic commands.

Reviewer: ```suggestion
For a list of supported magic commands, see [link-text](url).
```

![magic_sql](./1.2/magic_sql.gif)

### <a id="preconf-globals"></a>Preconfigured globals

> Just like on a Databricks cluster, we configure and provide you with some globals such as `spark` and `dbutils` when you run notebooks locally.

Reviewer: "we" is problematic (who exactly is "we"?).

> Just like on a Databricks cluster, we configure and provide you with some globals such as `spark` and `dbutils` when you run notebooks locally.

Reviewer: ```suggestion
Just like on a Databricks cluster, this update configures and provides you with some globals such as `spark` and `dbutils` when you run notebooks locally.
```

> For a list of supported globals, see [link-text](url).

Reviewer: I would leave this out, or link to the upcoming article in the tech docs.

Author: @PaulCornellDB I will wait for the docs to publish before we send this out. I will definitely want deep links everywhere.

![preconf_globals](./1.2/preconf_globals.gif)

### <a id="widgets"></a>Interactive `dbutils.widgets`

You can now use `dbutils.widgets` to create interactive widgets in your local notebooks.

> See [here]() for a list of requirements.

Reviewer: ```suggestion
For a list of requirements for interactive widgets, see [link-text](url).
```

![widgets](./1.2/widgets.gif)

> ### <a id="dbnb"></a>Support for Databricks Notebooks

Reviewer: ```suggestion
### <a id="dbnb"></a>Support for Databricks notebooks
```

> For a list of requirements for interactive widgets, see [link-text](url).

Reviewer: I don't think the upcoming article in the tech docs provides this list of requirements?

Author: Good catch. Can we add a section for this? It only needs 1 library. 

`pip install databricks-sdk[notebook]`

> You can use Databricks Notebooks locally, using [VS Code Interactive Windows](https://code.visualstudio.com/docs/python/jupyter-support-py).

Reviewer: ```suggestion
You can use Databricks notebooks locally with [VS Code Interactive Windows](https://code.visualstudio.com/docs/python/jupyter-support-py).
```

> See [here]() for more details.

Reviewer: ```suggestion
For more details, see [link-text](url).
```

Reviewer: "For more details" about what? Databricks notebooks? VS Code Interactive Windows? This feature?
